### DISCUSSION TITLE
does  tokio::io::copy support for pipe?

### DISCUSSION THREAD

**Discussion by ningmingxiao** [tokio-rs/tokio#6888] [Q&A]
some issue  [Tokio: task hangs when cancelling copy from a fifo](https://users.rust-lang.org/t/tokio-task-hangs-when-cancelling-copy-from-a-fifo-7/83946)
somebody said.
```
Operations on a tokio::fs::File are uncancelable because they are designed to work with real files. It is not designed for use with fifo pipes. Use an [UnixStream 9](https://docs.rs/tokio/latest/tokio/net/struct.UnixStream.html) for that.
```
I can't find any document about tokio::io::copy for pipe, Is it a bug?

**Comment by Darksonn** [comment]
Tokio has pipes here:

https://docs.rs/tokio/latest/tokio/net/unix/pipe/index.html

Of course they work with `tokio::io::copy`.

**Comment by ningmingxiao** [comment]
Does tokio tokio::io::copy  support os_pipe https://crates.io/crates/os_pipe  expect  tokio::net::unix::pipe ?
